### DISCUSSION TITLE
Difference between GET /api/artifacts/{id}/data/** and POST /api/artifacts/{id}/data

### DISCUSSION THREAD

**Discussion by ctoscano-eu** [International-Data-Spaces-Association/DataspaceConnector#124] [Q&A]
What is the major difference between these APIS:
     GET /api/artifacts/{id}/data/** 
     POST /api/artifacts/{id}/data
as both return the data related to the identified artifact.

**Comment by omarsilva1** [comment]
Hello César, the GET /api/artifacts/{id}/data/** endpoint allows the user to set additional query parameters, for instance: GET /api/artifacts/{id}/data/additional/path?param1=value1&param2=value2. 
These queries are sent to the provider connector and are used to get the data from the data source, limiting the scope of the data pulled.

More information about this can be found at: 
https://international-data-spaces-association.github.io/DataspaceConnector/CommunicationGuide/v6/Parametrization

**Reply by ctoscano-eu** [reply]
thank you @omarsilva1 for your explanation. 

The reason for the question is the following behaviour the DSC is having:
1- After the establishment of the agreement between a data provider and a data consumer, where the 'download' attribute is 'true',
doing both 'GET /api/artifacts/{id}/data/**' and 'POST /api/artifacts/{id}/data' triggers an IDS request artifact message from the consumer to the provider. which I think is the expected behaviour.
2- If the establishment of the agreement is done with the 'download' attribute as 'false', the 'POST /api/artifacts/{id}/data' doesn't trigger the  IDS request artifact message (which I think is the expected behaviour). However, the 'GET /api/artifacts/{id}/data/**' continues to trigger the  IDS request artifact message from the consumer to the provider, which I think is not the expected behaviour.
